lib/iris/experimental/xarray_dataset_wrapper.py

```diff
@@ -0,0 +1,415 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+"""
+A wrapper for an xarray.Dataset that simulates a netCDF4.Dataset.
+This enables code to read/write xarray data as if it were a netcdf file.
+
+NOTE: readonly, for now.
+TODO: add modify/save functions later.
+
+NOTE: this code is effectively independent of Iris, and does not really belong.
+However, this is a convenient place to test, for now.
+
+"""
+from collections import OrderedDict
+from typing import Optional
+
+import netCDF4 as nc
+import numpy as np
+import xarray
+import xarray as xr
+
+
+class _XrMimic:
+    """
+    An netcdf object "mimic" wrapped around an xarray object, which will be
+    either a dim, var or dataset.
```

Comment:
By dim do you mean DimCoord, in iris parlance?

Reply:
might be good to be explicit here: var I assume is `xr.Variable`, dataset is `xr.Dataset`. dim is an object that doesn't exist in xarray, but you need it for netcdf compat?